content/en/blog/_posts/2020-08-25-gsoc‘20 -building-operators-for-cluster-addons.md

```diff
@@ -0,0 +1,116 @@
+---
+layout: blog
+title: "GSoC 2020 - Building operators for cluster addons"
+date: 2020-08-25
+slug: gsoc20-building-operators-for-cluster-addons
+---
+
+**Author**: Onyekwere Somtochi
+
+# Introduction
+
+[Google Summer of Code](https://summerofcode.withgoogle.com/) is a global program that is geared towards introducing students to open source. Students are matched with open-source organizations to work with them for three months during the summer.
+
+My name is Somtochi Onyekwere from the Federal University of Technology, Owerri (Nigeria) and this year, I given the opportunity to work with Kubernetes (under the CNCF organization) and this led to an amazing summer spent learning, contributing and interacting with the community.
+
+Specifically, I worked on the [Cluster Addons: Package all things!](https://github.com/cncf/mentoring/blob/master/summerofcode/README.md#cluster-addons-package-all-the-things) project. The project focused on building operators for better management of various cluster addons, extending the tooling for building these operators and making the creation of these operators a smooth process.
+
+# Background
+
+Kubernetes has progressed greatly in the past few years with a flourishing community and a large number of contributors. The codebase has definitely moved away from the monolith structure where all the code resided in the [kubernetes/kubernetes](https://github.com/kubernetes/kubernetes) repository to being split into multiple sub-projects. Part of the focus of Cluster-addons is to make all these sub-projects work together in an easy to assemble, self-monitoring, self-healing and Kubernetes-native way. It enables them to work seamlessly without human intervention.
```

Comment:
suggesting small reword here:
> The codebase is gradually moving away from the monolith structure where all the code resides in the [kubernetes/kubernetes](https://github.com/kubernetes/kubernetes) repository...

Reply:
> Part of the focus of Cluster-addons is to make all these sub-projects work...

->

> Part of the focus of Cluster-addons is to make some of these sub-projects work...

(changed all -> some)

Reply:
^ did some rewording "all sub-projects" -> "some sub-projects"
also indicated that the move from monolith arch is still ongoing.